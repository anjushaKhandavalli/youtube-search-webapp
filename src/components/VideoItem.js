import React from 'react';
import { VideoItemContainer, VideoInfo, Thumbnail, Title, Description } from './VideoItem.style';

const VideoItem = ({video}) => {
  return (
    <VideoItemContainer data-testid="video-item">
        <Thumbnail data-testid="thumbnail" src={video.snippet.thumbnails.medium.url} alt={video.snippet.description}/>
        <VideoInfo>
            <Title data-testid="title">{video.snippet.title}</Title>
            <Description data-testid="description">{video.snippet.description}</Description>
        </VideoInfo>
    </VideoItemContainer>
)
}

export default VideoItem;