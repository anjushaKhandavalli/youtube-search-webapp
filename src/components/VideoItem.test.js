import React from 'react';
import { render } from "@testing-library/react";
import VideoItem from "./VideoItem";
const mockVideo = {
  id: {
    kind: "youtube#video",
    videoId: "video_1"
  },
  snippet: {
    title: "video title",
    description: "description about the video",
    thumbnails: {
      medium : {
        url: "https://i.ytimg.com/vi/AETFvQonfV8/default.jpg",
      }
    }
  }
}
describe("VideoItem", () => {
  it("should render the video item with thumbnail, title and description",() => {
    const {container, getByTestId } = render(<VideoItem video={mockVideo}/>)

    expect(container).toMatchSnapshot();
    expect(getByTestId('thumbnail')).toBeDefined();
    expect(getByTestId('title')).toBeDefined();
    expect(getByTestId('description')).toBeDefined();
  })
})