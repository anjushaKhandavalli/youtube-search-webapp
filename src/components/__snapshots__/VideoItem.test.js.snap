// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoItem should render the video item with thumbnail, title and description 1`] = `
<div>
  <div
    class="sc-bdfBwQ jwCXbJ"
    data-testid="video-item"
  >
    <img
      alt="description about the video"
      class="sc-gsTCUz dLvDma"
      data-testid="thumbnail"
      src="https://i.ytimg.com/vi/AETFvQonfV8/default.jpg"
    />
    <div
      class="sc-dlfnbm jxTFsF"
    >
      <h2
        class="sc-hKgILt ksesVL"
        data-testid="title"
      >
        video title
      </h2>
      <h4
        class="sc-eCssSg gqHjlA"
        data-testid="description"
      >
        description about the video
      </h4>
    </div>
  </div>
</div>
`;
